src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -154,6 +141,12 @@
            (str " at " file ":" (:line rule)))
          (format " (authentic: %s)" authentic?))))
 
+(schema/defn sort-rules :- Rules
+  "Sorts the rules based on their :sort-order, and then their :name if they
+   have the same sort order value."
+  [rules :- Rules]
+  (vec (sort-by (juxt :sort-order :name) rules)))
```

Comment:
Oh right, I made it a vector because of the input+output schema `[Rule]`.

But, as I discovered recently, the schema `[Foo]` doesn't require the collection to be a vector, just something sequential.  So stripping off `vec` here satisfies the schema and everything works fine.  I was surprised when I discovered schema works this way, but I think it makes sense.